torch/csrc/jit/script/python_sugared_value.cpp

```diff
@@ -302,6 +305,42 @@ std::shared_ptr<SugaredValue> ModuleValue::attr(
     auto list = g.insertNode(g.createTuple(params))->output();
     return std::make_shared<ConstantParameterList>(list);
   }
+
+  // If recursive script mode is on, create a ScriptModule and register it as
+  // as submodule or register a python method as a script::Method
+  if (getRecursiveScriptMode()) {
+    if (py::isinstance(attr, py::module::import("torch.nn").attr("Module"))) {
+      // If the module is a submodule of the py_module, convert it to a
```

Comment:
In our discussions, we always treated `script` as making a copy of a module. But here we are clearly mutating a submodule. I am not sure how I feel about this behavior because of that.

Reply:
How is the submodule being mutated? It's being passed on to `_convert_to_script_module()` which shouldn't be touch the original module at all.